platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java

```diff
@@ -120,38 +120,51 @@ public void onReceived(ObservedPayment payment) {
                         .build()))
             .build();
 
+    // Update statuses
     TransactionEvent.Status oldStatus = TransactionEvent.Status.from(txn.getStatus());
-    TransactionEvent.Status newStatus;
-    String message;
+    TransactionEvent.Status newStatus = TransactionEvent.Status.PENDING_RECEIVER;
 
     // Check if the payment contains the expected amount (or greater)
     BigDecimal expectedAmount = decimal(txn.getAmountIn());
     BigDecimal gotAmount = decimal(payment.getAmount());
-    if (gotAmount.compareTo(expectedAmount) < 0) {
-      updateTransactionStatusTo(ERROR, txn, payment, stellarTransaction);
-
-      newStatus = TransactionEvent.Status.ERROR;
+    String message = "Incoming payment for SEP-31 transaction";
+    ;
+    if (gotAmount.compareTo(expectedAmount) >= 0) {
+      Log.info(message);
+      txn.setTransferReceivedAt(paymentTime);
```

Comment:
So we're not setting the transfer receive time if the payment was insufficient?

Reply:
Yeah, my rationale is the `transferReceivedAt` would indicate the expected transfer was received, but it actually was not received with the expected specs.

I'm happy to discuss it further.

Reply:
My thought is that there is no harm in saving the time the payment was received even if it was insufficient, then we can overwrite the value if another transaction with the same memo sends the remaining amount.

But I don't feel strongly.